### PR TITLE
Improve error messages from Terraform bats tests

### PR DESCRIPTION
**Pull request by Nuru**
## what
- Improve error messages from Terraform bats tests

## why
- Output was not understandable

### CODE REVIEW

File: test/terraform/provider-pinning.bats [hunk 1]
```diff
@@ -12,22 +16,22 @@ function teardown() {
   unset TF_CLI_ARGS_init
 }
 
-@test "check if terraform-config-inspect is installed" {
```
Comment (osterman) [inline test/terraform/provider-pinning.bats:15]: Why are we deleting this?

File: test/terraform/module-pinning.bats [hunk 2]
```diff
@@ -1,6 +1,10 @@
 load 'lib'
 
 function setup() {
+  if ! which terraform-config-inspect; then
```
Comment (osterman) [inline test/terraform/module-pinning.bats:4]: Note we use the other pattern for more than config inspect. We should then be consistent.

**Review by osterman — APPROVED** [review]

File: test/terraform/validate.bats [hunk 3]
```diff
@@ -4,7 +4,7 @@ function setup() {
   skip_unless_terraform
   clean
   export TF_CLI_ARGS_init="-input=false -backend=false"
-  terraform init
+  terraform init >/dev/null
```
[the remark this answers is not in the document]
Comment (Nuru) [reply test/terraform/validate.bats:7]: `terraform init` outputs a lot of progress information, like this:
```
Initializing modules...
Downloading git::https://github.com/cloudposse/terraform-aws-dynamodb-autoscaler.git?ref=tags/0.9.0 for dynamodb_autoscaler...
- dynamodb_autoscaler in .terraform/modules/dynamodb_autoscaler
Downloading cloudposse/label/null 0.22.0 for dynamodb_autoscaler.this...
- dynamodb_autoscaler.this in .terraform/modules/dynamodb_autoscaler.this
Downloading cloudposse/label/null 0.22.0 for this...
- this in .terraform/modules/this

Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "aws" (hashicorp/aws) 3.21.0...
- Downloading plugin for provider "null" (hashicorp/null) 3.0.0...
```

and `bats` was not suppressing it even on success. It was making the output of `bats` hard to read/parse/understand and added negligible value. If there is an error, we will still see it on `stderr`.

File: test/terraform/provider-pinning.bats (hunk 1 above)
Comment (Nuru) [reply test/terraform/provider-pinning.bats:15]: It is not a test of the PR/code, it is a test of the test harness. It should not be called out as and `ok` or `not ok` test of the PR.

File: test/terraform/module-pinning.bats (hunk 2 above)
Comment (Nuru) [reply test/terraform/module-pinning.bats:4]: The `bats` output is a numbered list of tests with `ok` or `not ok`. Those should be tests of the code, not tests of the test harness. This pattern of testing in the setup will still cause the test to fail with an error message, but does not add to the number of tests or give an `ok`/`not ok` on the code being tested. If we are using `@test` elsewhere to test the test harness, then yes, we should get rid of those, too.

**Pull request merged**